lib/protoparser/influx/parser.go

```diff
@@ -176,47 +184,50 @@ func (f *Field) unmarshal(s string, noEscapeChars, hasQuotedFields bool) error {
 	return nil
 }
 
-func unmarshalRows(dst []Row, s string, tagsPool []Tag, fieldsPool []Field) ([]Row, []Tag, []Field) {
+func (rs *Rows) unmarshal(s string) error {
 	noEscapeChars := strings.IndexByte(s, '\\') < 0
 	for len(s) > 0 {
 		n := strings.IndexByte(s, '\n')
 		if n < 0 {
 			// The last line.
-			return unmarshalRow(dst, s, tagsPool, fieldsPool, noEscapeChars)
+			return rs.unmarshalRow(s, noEscapeChars)
+		}
+		err := rs.unmarshalRow(s[:n], noEscapeChars)
+		if err != nil && !rs.IgnoreErrs {
+			return err
```

Comment:
I think, it's better to handle error at single place.
E.g.:

```go
err := rs.unmarshalRow(s[:n], noEscapeChars)
		if err != nil{
                   if !rs.IgnoreErrs {
                       return fmt.Errorf("incorrect influx line %q: %w",s,err )
                    }
	      	  logger.Errorf("skipping InfluxDB line %q because of error: %s", s, err)
		 invalidLines.Inc()
              } 
```

 It shouldn't trigger additional metric and produce a single meaningful error.

Reply:
done